core/update.php

```diff
@@ -304,7 +304,12 @@ function update_info_page() {
     $output .= $module_status_report;
   }
 
-  $form_action = check_url(backdrop_current_script_url(array('op' => 'check_updates', 'token' => $token)));
+  // Go straight to update selection if upgrading from Drupal 7.
+  $op = (backdrop_get_installed_schema_version('system') > 7000) ? 'selection' : 'check_updates';
```

Comment:
Isn't `state_get('update_d7_upgrade')` available already? Not sure - didn't actually test it...

Reply:
No, I tried that first but it's set later. We could potentially set that earlier and use it though.

Reply:
See what you think. I also tested locally using the state in a few places in `core/includes/update.inc` where the same `backdrop_get_installed_schema_version('system') > 7000` check is done, but that seems to cause issues with the upgrade so I reverted those and left those out rather than open a can of worms.